src/components/CustomToast/CustomToast.tsx

```diff
@@ -0,0 +1,136 @@
+import "./CustomToast.scss";
+import {FC, useEffect, useRef, useState} from "react";
+import {ReactComponent as CloseIcon} from "assets/icon-close.svg";
+import {ToastTypes} from "utils/Toast";
+import classNames from "classnames";
+
+export interface CustomToastProps {
+  title: string;
+  message?: string;
+  hintMessage?: string;
+  hintOnClick?: () => void;
+  buttons?: string[];
+  firstButtonOnClick?: () => void;
+  secondButtonOnClick?: () => void;
+  icon?: React.FunctionComponent<React.SVGProps<SVGSVGElement>>;
+  iconName?: string;
+  type?: ToastTypes;
+}
+
+export const CustomToast: FC<CustomToastProps> = ({title, message, buttons, hintMessage, hintOnClick, firstButtonOnClick, secondButtonOnClick, icon, iconName, type}) => {
+  const [isSingleToastTitle, setIsSingleToastTitle] = useState<boolean>(true);
+  const [isMobile, setIsMobile] = useState<boolean>(false);
+  const titleRef = useRef<HTMLDivElement>(null);
+  const standardIcon = ["info", "success", "error"].includes(iconName!);
+  const Icon = icon;
+
+  const titleMaxHeightDesktop = 38; // two lines of text
+  const titleMaxHeightMobile = 19; // one line of text
+  const maxWidthMobile = 767; // $smartphone: "screen and (max-width: 767px)"
+
+  // check whether screensize is mobile/desktop
+  useEffect(() => {
+    const handleResize = () => {
+      setIsMobile(window.innerWidth <= maxWidthMobile);
+    };
+    window.addEventListener("resize", handleResize);
+    handleResize();
+    return () => window.removeEventListener("resize", handleResize);
+  }, []);
+
+  // detects whether the title spans one or two lines
+  useEffect(() => {
+    if (!titleRef.current) {
+      return;
+    }
+    if (!isMobile) {
+      if (titleRef.current.offsetHeight > titleMaxHeightDesktop) {
+        setIsSingleToastTitle(false);
+      }
+    }
+    if (isMobile) {
+      if (buttons) {
+        if (buttons.length < 2) {
+          if (titleRef.current.offsetHeight > titleMaxHeightMobile) {
+            setIsSingleToastTitle(false);
+          }
+        } else {
+          setIsSingleToastTitle(false);
+        }
+      } else if (titleRef.current.offsetHeight > titleMaxHeightDesktop) {
+        setIsSingleToastTitle(false);
```

Comment:
this whole if branch tree looks like it can be simplified to be more readable, but I'm not sure